include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -28,10 +28,34 @@ class NBL_API IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescrip
 		using base_t = asset::IDescriptorSet<const IGPUDescriptorSetLayout>;
 
 	public:
-		IGPUDescriptorSet(core::smart_refctd_ptr<const ILogicalDevice>&& dev, core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& _layout) : base_t(std::move(_layout)), IBackendObject(std::move(dev)) {}
+		IGPUDescriptorSet(core::smart_refctd_ptr<const ILogicalDevice>&& dev, core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& _layout, core::smart_refctd_ptr<IDescriptorPool>&& pool, const uint32_t* descriptorStorageOffsets)
+			: base_t(std::move(_layout)), IBackendObject(std::move(dev)), m_pool(std::move(pool))
+		{
+			memcpy(m_descriptorStorageOffsets, descriptorStorageOffsets, asset::EDT_COUNT * sizeof(uint32_t));
+		}
+
+		uint8_t* getDescriptorMemory(const asset::E_DESCRIPTOR_TYPE type, const uint32_t binding) const;
```

Comment:
needs to be reworked into the [Dependency Injection](https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#issuecomment-1263217433)

Reply:
Resolved.

Reply:
yes resolved, DI was a mistake. whats there right now is fine